pkg/apis/kubeflow.org/v2alpha1/trainjob_types.go

```diff
@@ -213,15 +224,19 @@ type OutputModel struct {
 // PodSpecOverride represents the custom overrides that will be applied for the TrainJob's resources.
 type PodSpecOverride struct {
 	// TrainJobs is the training job replicas in the training runtime template to apply the overrides.
+	// +listType=map
```

Comment:
```suggestion
	// +listType=atomic
```

Could we use atomic type since we may convert the name field to optional, then introduce selectors.